SYCL/ESIMD/api/functional/functions/functions_atomic_update_inc_dec.cpp

```diff
@@ -0,0 +1,230 @@
+                work_item_index);
+          });
+    });
+    queue.wait_and_throw();
+
+    std::sort(shared_result.begin(), shared_result.end());
+
+    ChangeElemFilterT filter{};
+    size_t num_changed_elems = 0;
+    for (size_t i = 0; i < NumElemsToChange; ++i) {
+      if (filter(i) == 1) {
+        ++num_changed_elems;
+      }
+    }
+
+    for (size_t i = 0; i < NumElems; ++i) {
+      const DataT &retrieved = shared_result[i];
+      if (i < NumberIteractions) {
+        if (result_fill_value != retrieved) {
+          passed = fail_test(i, result_fill_value, retrieved, data_type,
+                             "atomic_update return value");
+        }
+      } else {
+        const DataT &expected = base_value * num_changed_elems +
+                                (i - NumberIteractions) * num_changed_elems;
+        if (expected != retrieved) {
+          passed = fail_test(i, expected, retrieved, data_type,
+                             "atomic_update return value");
+        }
+      }
+    }
+
+    std::vector<size_t> selected_indexes;
+    // Collect the indexess that has been selected.
+    for (size_t i = 0; i < NumElemsToChange; ++i) {
+      if (filter(i) == 1) {
+        selected_indexes.push_back(indexes_to_change[i]);
+      }
+    }
+
+    // Push the largest value to avoid the following error: can't dereference
+    // out of range vector iterator.
+    selected_indexes.push_back(std::numeric_limits<size_t>::max());
+    auto next_selected_index = selected_indexes.begin();
+
+    const DataT &expected = base_value;
+    DataT expected_after_change = functions::get_expected_value<ChosenOperator>(
+        base_value, NumberIteractions);
+    // Verify that values, that do not was changed has initial values.
+    for (size_t i = 0; i < NumElems; ++i) {
+      // If current index is less than selected index verify that this element
+      // hasn't been changed.
+      const DataT &retrieved = shared_init_values[i];
+      if (i < *next_selected_index) {
+        if (expected != retrieved) {
+          passed = fail_test(i, expected, retrieved, data_type,
+                             "value that should be updated");
+        }
+      } else {
+        if (expected_after_change != retrieved) {
+          passed = fail_test(i, expected_after_change, retrieved, data_type,
+                             "value that should be updated");
+        }
+        next_selected_index++;
+      }
+    }
+
+    return passed;
+  }
+
+private:
+  bool fail_test(size_t i, DataT expected, DataT retrieved,
+                 const std::string &data_type,
+                 const std::string &verification_type) {
+    log::fail(TestDescriptionT(data_type), "Unexpected value at index ", i,
+              ", retrieved: ", retrieved, ", expected: ", expected,
+              ", for verification type: ", verification_type,
+              ", with number elements to change: ", NumElemsToChange);
```

Comment:
It's up to you, but this one looks like a test case description detail.

Reply:
maybe, but I suggest use this way now and leave the option to change it in the future